src/cattrs/_compat.py

```diff
@@ -203,7 +210,12 @@ def get_final_base(type) -> Optional[type]:
         from typing import _LiteralGenericAlias
 
         def is_literal(type) -> bool:
-            return type.__class__ is _LiteralGenericAlias
+            return type in LITERALS or (
+                isinstance(
+                    type, (_GenericAlias, _LiteralGenericAlias, _SpecialGenericAlias)
+                )
+                and type.__origin__ in LITERALS
+            )
```

Comment:
this is isomorphic to what `typing-inspect` does and it works in our company's testing as well.